cdap-ui/app/cdap/components/ConfirmationModal/index.js

```diff
@@ -25,6 +25,9 @@ require('./ConfirmationModal.scss');
 export default class ConfirmationModal extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      headerTitle: this.props.headerTitle
```

Comment:
I did it this way because I didn't want the header title to change when new props come in, which was causing the bug where the 'Stop Program' label changed to 'Start Program' when the status was loading. I agree that it wasn't very clean, so I've reverted this change, and changed the implementation of `StartStopAction` instead.